submissions/nvklap/dom/js/app.js

```diff
@@ -0,0 +1,407 @@
+      const li = document.createElement('li');
+      li.classList.add('series-item');
+      li.append(link);
+      ul.append(li);
+    });
+  navWrapper.append(ul);
+}
+
+function toggleMenu() {
+  const btnImg = document.querySelector('.btn-img');
+  if (navList.className.includes('show')) {
+    navList.classList.remove('show');
+    btnImg.classList.remove('open');
+  } else {
+    navList.classList.add('show');
+    btnImg.classList.add('open');
+  }
+}
+
+function createContentImg(src, alt) {
+  const img = document.createElement('img');
+  img.classList.add('content-img');
+  img.setAttribute('src', src);
+  img.setAttribute('alt', alt);
+  const figure = document.createElement('figure');
+  figure.classList.add('content-figure');
+  figure.append(img);
+  return figure;
+}
+
+function createContentHeading(tag, className, text) {
+  const heading = document.createElement(tag);
+  heading.classList.add(className);
+  heading.textContent = text;
+  return heading;
+}
+
+function createContentList(castArr) {
+  const ul = document.createElement('ul');
+  ul.classList.add('cast-list');
+  castArr.forEach((player) => {
+    player = player.split(' ');
+    let [firstName, ...lastName] = player;
+    lastName = lastName.join(' ');
+    const span = document.createElement('span');
+    span.classList.add('red');
+    span.textContent = firstName;
+    const li = document.createElement('li');
+    li.classList.add('cast-item');
+    li.append(span);
+    li.append(lastName);
+    ul.append(li);
+  });
+  return ul;
+}
+
+function createContentText(paragraphsArr) {
+  const paragraphsWrapper = document.createElement('div');
+  paragraphsArr.forEach((text) => {
+    const pTag = document.createElement('p');
+    pTag.classList.add('content-text');
+    pTag.textContent = text;
+    paragraphsWrapper.append(pTag);
+  });
+  return paragraphsWrapper;
+}
+
+function createContentBody(title, series, cast, text) {
+  const contentTitle = createContentHeading('h2', 'content-title', title);
+  const contentSubtitle = createContentHeading(
+    'h3',
+    'content-subtitle',
+    series
+  );
+  const contentCastList = createContentList(cast);
+  const contentTexts = createContentText(text);
+
+  const contentBody = document.createElement('div');
+  contentBody.classList.add('content-body');
+  contentBody.append(
+    contentTitle,
+    contentSubtitle,
+    contentCastList,
+    contentTexts
+  );
+  return contentBody;
+}
+
+function createContent({ img, alt, title, series, cast, text }) {
+  const contentImg = createContentImg(img, alt);
+  const contentBody = createContentBody(title, series, cast, text);
+  return [contentImg, contentBody];
+}
+
+function addTemplateToDom(parent, template) {
+  parent.innerHTML = '';
+  parent.append(...template);
+}
+
+function initContent(parent, initSeriesData) {
+  const htmlTemplate = createContent(initSeriesData);
+  addTemplateToDom(parent, htmlTemplate);
+}
+
+function showContent({ target }) {
```

Comment:
`addNavigation`, `createContentList` and `showContent` do too much stuff. Try to decompose them into smaller functions.